### PR TITLE
Fix issues with QuotedMacroType

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/scalafmt/issues/2512

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ import org.portablescala.sbtplatformdeps.PlatformDepsPlugin.autoImport._
 
 object Dependencies {
   val metaconfigV = "0.9.10"
-  val scalametaV  = "4.4.17"
+  val scalametaV  = "4.4.18"
```
Comment (kitbellew) [inline project/Dependencies.scala:9]: i'd perhaps recommend upgrading as a separate, explicit PR

**Review by kitbellew — APPROVED** [review]

**Review by kitbellew — CHANGES_REQUESTED** [review]
found another pr with the same upgrade...

**Comment by tgodzik** [comment]
> found another pr with the same upgrade...

I can put it into a different one, but I am not sure it's that useful in this case. It doesn't break any tests.

**Comment by kitbellew** [comment]
> > found another pr with the same upgrade...
> 
> I can put it into a different one, but I am not sure it's that useful in this case. It doesn't break any tests.

of course, but in this case `git log` will not tell you when you've upgraded the version (and possibly caused something else).

**Pull request merged**